### ISSUE TITLE
Improve Accessibility under NVDA Screen Reader

### ISSUE THREAD

**Issue by martinwoodward** [OpenLiveWriter/OpenLiveWriter#394]
None of the edit boxes read read any text at all under the common (free) [NVDA](http://www.nvaccess.org/) reader. Be great to improve how OLW works with readers.

**Comment by rkingett** [comment]
The good thing is if it works with NVDA it will work with all other readers, too!

**Comment by kathweaver** [comment]
So what is the issue?  I've downloaded and gone through most of Open Live Writer and haven't seen an issue on how it works.

**Comment by rkingett** [comment]
NVDA does not read any of the edit fields in the main window, when you are
writing a blog post. NVDA reads all the menus and otherwise just fine but
not text that you wrote in the main edit fields. Title, or body

Robert Kingett*, Journalist <https://blindjournalist.wordpress.com/resume/>*

*My Patreon* <https://www.patreon.com/blindwriter>

*Other ways to contact me*
<https://blindjournalist.wordpress.com/contact-this-blogger/>

**Comment by kathweaver** [comment]
Thanks but on my machine, when I type in the entry it reads the letters as I go.  When I stop, it reads what I have written so far.

If I put my mouse on the title, it reads the title to me.

If I put my mouse on the edit entry, it reads the entry to me.

It also reads the preview and source to me.

Now, when it is finished, it's a bit inconsistent, but I can get it to read everything.

**Comment by rkingett** [comment]
And that is the problem, the fact you are using the mouse. Try arrowing
through, and hit shift tab to go to the title edit box. Blind or VI people
use keyboard commands instead of a mouse.

Robert Kingett*, Journalist <https://blindjournalist.wordpress.com/resume/>*

*My Patreon* <https://www.patreon.com/blindwriter>

*Other ways to contact me*
<https://blindjournalist.wordpress.com/contact-this-blogger/>

**Comment by kathweaver** [comment]
Thanks for clarifying and I understand what the problem is now.  Fixing <shrug>

**Comment by rkingett** [comment]
Thanks! If you have any more questions, feel free to ask. :)

Robert Kingett*, Journalist <https://blindjournalist.wordpress.com/resume/>*

*My Patreon* <https://www.patreon.com/blindwriter>

*Other ways to contact me*
<https://blindjournalist.wordpress.com/contact-this-blogger/>

**Comment by kathweaver** [comment]
I think I know where the issue lies, but I can't open the proper forms in Visual Studio.

When I go to OpenLiveWriter.PostEditor.Form (and some others), I get: 

Unexpected Error Occurred

System.NullReferenceException: Object reference not set to an instance of an object.
   at OpenLiveWriter.ApplicationFramework.SatelliteApplicationForm.PositionMainControl() in C:\Users\kathw_000\Documents\GitHub\OpenLiveWriter\src\managed\OpenLiveWriter.ApplicationFramework\SatelliteApplicationForm.cs:line 168
   at OpenLiveWriter.ApplicationFramework.SatelliteApplicationForm.OnInitializeWorkspace() in C:\Users\kathw_000\Documents\GitHub\OpenLiveWriter\src\managed\OpenLiveWriter.ApplicationFramework\SatelliteApplicationForm.cs:line 144
   at OpenLiveWriter.ApplicationFramework.SatelliteApplicationForm.OnLoad(EventArgs e) in C:\Users\kathw_000\Documents\GitHub\OpenLiveWriter\src\managed\OpenLiveWriter.ApplicationFramework\SatelliteApplicationForm.cs:line 76

**Comment by rkingett** [comment]
I wish I could help but do not know code! I passed this thread off to a few developers in the blind community, though

**Comment by kathweaver** [comment]
I figure one of the experienced devs will see this and sort it out.  The good news is that at least I know where to start, and it's going to be some meticulous work but its doable.

**Comment by rkingett** [comment]
I wonder now, if windows live writer has this same issue? Back then I was using JAWS, which got every information through a driver that JAWS uses. Now, I strictly use NVDA because, basically, if things work with NVDA they will work with any and every screen reader.

I used WLW with JAWS with no problem but I think that was because JAWS was pulling information from this display mirror driver. NVDA just reads from the main code, so YAY.

The good thing is, after this is all done, it will not have to be done again!

**Comment by kathweaver** [comment]
Yes, probably.  The reason it isn't working with NVDA is that a lot of the form's accessibility fields haven't been filed out.  They probably weren't filled out with WLW either.

**Comment by rkingett** [comment]
As far as I can tell, options edit boxes read the names of the edit boxes. I will need to go back and see if the blog edit boxes read, you know, the dialog box that has you enter your blog info? The one in the settings menu reads just fine but right now the 2 edit fields that are not reading are the compose edit and the title edit

**Comment by kathweaver** [comment]
Right.  I've gone through some of the forms and dialog boxes, and there is nothing filled out in the accessibility properties.  So someone (probably me, once I figure how to open all the forms), needs to go through and fill out all the accessibility properties.

**Comment by kathweaver** [comment]
#rkingett Windows Forms and Dialogs boxes are made with (among other methods) a markup language called XAML.  Part of that language is some accessibility properties.

**Comment by lextm** [comment]
@kathweaver XAML is not used in Windows Forms. XAML was designed for WPF, and later used also in Silverlight, Windows Store apps/UWP, Xamarin.Forms.

`SatelliteApplicationForm` is not visual designer friendly, as it requires some control to be initialized. Without fixing that (as least for the visual designer to run), all changes of it and its derived forms must be done manually.

**Comment by kathweaver** [comment]
Okay, thanks for clearing that up.

**Comment by flcdrg** [comment]
@kathweaver https://msdn.microsoft.com/en-us/library/cb35a5fw(v=vs.110).aspx has some useful info on WinForms accessibility

**Comment by kathweaver** [comment]
I have thought about doing this at runtime, but it really needs to be done at design time for documentation purpose if nothing else so we really need #403 before this can be fixed.

**Comment by rkingett** [comment]
when the program becomes more accessible I can write up documentation and keyboard shortcuts about accessibility and screen readers and otherwise. OLW works with high contrast very well, so YAY! Thanks all!

I did find a few small spelling errors on the website and would like to fix thhem. How can I do this on the web?

**Comment by timheuer** [comment]
Given the edit area is HTML edit this is probably an issue with hosting the component and the UIA tree not being 'seen' either.

**Comment by kathweaver** [comment]
If you see spelling errors on the website, go to the GitHub for it https://github.com/OpenLiveWriter/OpenLiveWriter.Github.io  and we'll get them fixed. I already know about one.

**Comment by kathweaver** [comment]
I was hoping this was something I could work on but the frustration factor has overwhelmed me.

**Comment by rkingett** [comment]
Darn! I passed this off to a few blind developers on Twitter and otherwise and so I hope a few jump in and help with this. So far, no bites, but I will keep sharing this thread!

**Comment by rkingett** [comment]
Is there a text editor that NVDA can access via the keyboard?

**Comment by rkingett** [comment]
I passed this thread off to a few developers so hopefully they can help with making this accessible to screen readers.

**Comment by rkingett** [comment]
Hey all! I would love to be able to use this client but I still can't use it fully. Is there any plan to work on screen reader access? NVDA would be the best to test with.

**Comment by rkingett** [comment]
what about adding a temporary screen reader mode where a simple text box is used instead of the normal editor until the main one can be fully accessible?

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
